### PR TITLE
Add colour to posts

### PR DESCRIPTION
**Pull request by dNitza**
This PR adds color classes to posts and cleans up the post edit/creation forms.

screen shot 2016-05-19 at 9 26 19 am
screen shot 2016-05-19 at 9 30 57 am

* A post can not have the same color as one of the 5 (can change) posts before it.
* Colors are picked from a pool of predetermined colors defined in `Types::PostHighlightColors`.

### CODE REVIEW

**Comment by makenosound** [comment]
@dNitza You can split each row into even columns by using `group`, and you can use a `section` to collect a set of fields together so there’s a bit more structure, but at the moment there’s no support for more complex layouts.

```ruby
section :section, label: "Section label" do
  text_field :section_text_field,
    label: "Section text field"

  number_field :section_number_field,
    label: "Section number field"

  group label: "Group label" do
    text_field :group_text_field,
      label: "Group text field",
      validation: {
        filled: true
      }
    number_field :group_number_field,
      label: "Group number field"
  end
end
```

**Comment by dNitza** [comment]
Oh nice, thanks for that @makenosound !

File: lib/types.rb [hunk 1]
```diff
@@ -2,4 +2,6 @@
 
 module Types
   include Dry::Types.module
+
+  Color = Types::Strict::String.enum("blue", "yellow", "red")
```
Comment (timriley) [inline lib/types.rb]: Probably best to rename this to `PostHighlightColor` or something?

File: apps/admin/lib/admin/entities/post.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ class Post < Dry::Types::Value
       attribute :status, Status
       attribute :author_id, Types::Strict::Int
       attribute :published_at, Types::Time
+      attribute :color, Types::Strict::String
```
Comment (timriley) [inline apps/admin/lib/admin/entities/post.rb]: This can use the same type that you define in `module Types` :)

Comment (dNitza) [inline apps/admin/lib/admin/entities/post.rb]: Oh right! I think I remember seeing that. Will it complain if we try and set a color not set in the Type?

Comment (timriley) [inline apps/admin/lib/admin/entities/post.rb]: Yep, anything outside the type definition will be invalid and prevent the entity from initialising

**Pull request merged**